server/utils/removeIndexCollisions.js

```diff
@@ -0,0 +1,64 @@
+// @flow
+import fractionalIndex from "fractional-index";
+import { Collection } from "../models";
+import { sequelize } from "../sequelize";
+
+/**
+ *
+ * @param teamId The team id whose collections has to be fetched
+ * @returns An array or undefined. If array, each array element is an array of collided collection id and its new index [id,newIndex]
+ */
+export default async function removeIndexCollisions(teamId: string) {
+  let collections = await Collection.findAll({
```

Comment:
This is still a bit brute-force for my liking, we really shouldn't need to query _all_ collections like this as we're always constraining collisions to a collection that's being added or updated.

Given that this is an infrequent operation I don't think it should hold up merging the PR though

Reply:
I am pushing an update, pls bear with me.